Crash caused by angelsindustry / angelsrefining

**Describe the bug**
A clear and concise description of what the bug is.

```
I am developing a mod called Atomic Overhaul and a user made me aware of a crash during startup when playing with my mod and Angelsindustry. I have no idea how to fix it and so I wanted to ask for help.
```

**To Reproduce**
Information to reproduce the behavior:
1. Game Version
```
Everything is Up to date (Factorio and all angels mods)
```
2. Modlist
```
Atomic Overhaul
Angels Industry (with all of its dependencies)
```

**Additional context**
Add any other context about the problem here.
```
2.486 Error ModManager.cpp:1558: Failed to load mod "angelsindustries": __angelsrefining__/prototypes/recipe-builder.lua:217: attempt to compare number with nil
stack traceback:
	__angelsrefining__/prototypes/recipe-builder.lua:217: in function 'add'
	__angelsrefining__/prototypes/recipe-builder.lua:429: in function 'p_merge_item_lists'
	__angelsrefining__/prototypes/recipe-builder.lua:509: in function 'p_result_merge'
	__angelsrefining__/prototypes/recipe-builder.lua:546: in function 'p_copy'
	__angelsrefining__/prototypes/recipe-builder.lua:567: in function 'patch'
	__angelsrefining__/prototypes/override-functions.lua:968: in function 'execute'
	...lsindustries__/prototypes/angels-industries-override.lua:32: in main chunk
	[C]: in function 'require'
	__angelsindustries__/data-updates.lua:5: in main chunk
   2.487 Loading mod core 0.0.0 (data.lua)
   2.586 Checksum for core: 1476961332
   2.633 Error ModManager.cpp:1558: Error in assignID: recipe-category with name 'crafting' does not exist.
```

Sorry for my bad grammar, English is not my native language.

--- a/angelsrefining/changelog.txt
+++ b/angelsrefining/changelog.txt
@@ -22,6 +22,8 @@ Date: ???
       - Made some mineral sludge crystallization recipe require Crystallizer 2
       - Made crystal catalyst recipe require Crystallizer 2
       - Made hybrid catalyst recipe require Crystallizer 3
+  Bugfixes:
+    - Fixed error when patching a recipe with variable result products (835)
 ---------------------------------------------------------------------------------------------------
 Version: 0.12.2
 Date: 06.06.2022

--- a/angelsrefining/prototypes/recipe-builder.lua
+++ b/angelsrefining/prototypes/recipe-builder.lua
@@ -426,7 +426,11 @@ local function p_merge_item_lists(base, patch)
       init()
       for _, item in pairs(base) do
          local n, t, a = item.name or item[1], item.type or "item", item.amount or item[2]
-         add(n, t, a)
+         if a == nil then
+            add2(n, t, item.amount_min or -1, item.amount_max or -1)
+         else
+            add(n, t, a)
+         end
          adjust_secondary_atts(n, t, item)
       end
       for _, item in pairs(patch) do

